dotnet/src/Agents/Abstractions/AgentChat.cs

```diff
@@ -179,6 +179,12 @@ await foreach (var message in channel.InvokeAsync(agent, cancellationToken).Conf
                 this.History.Add(message);
                 messages.Add(message);
 
+                if (message.Role == AuthorRole.Tool || message.Items.All(i => i is FunctionCallContent))
```

Comment:
Is there a reason to filter out function call content and results? What I've noticed is that if both are excluded from a chat history and the chat history is sent to LLM again, the scenario that is possible when the whole chat is modeled as an agent, the LLM will call the functions again.

Reply:
This is only filtering out what is returned to the user.  The content is included in the history.

This is intended to align with core SK behavior